File: 125.valid-palindrome.py
#
# @lc app=leetcode id=125 lang=python3
#
# [125] Valid Palindrome
#

# @lc code=start
import re

class Solution:
    def isPalindrome(self, s: str) -> bool:
        # fmt_string = re.sub(r'[^a-zA-Z0-9]', '', s.lower())
        fmt_string = ""
        for c in s.lower():
            if c.isalnum():
                fmt_string += c 

        str_len = len(fmt_string)
        for i in range(0, str_len//2):
            if fmt_string[i] != fmt_string[str_len - 1 - i]:
                return False

        return True
        
# @lc code=end

File: 125.valid-palindrome.rs
/*
 * @lc app=leetcode id=125 lang=rust
 *
 * [125] Valid Palindrome
 */

// @lc code=start
impl Solution {
    pub fn is_palindrome(s: String) -> bool {
        let mut char_array: Vec<char> = Vec::new();
        for c in s.to_ascii_lowercase().chars(){
            if c.is_alphanumeric(){
                char_array.push(c)
            }
        }

        for i in 0..char_array.len()/2 {
            if char_array[i] != char_array[char_array.len() - 1 - i]{
                return false;
            }
        }

        true
    }
}
// @lc code=end

